addons/survey/models/survey_survey.py

```diff
@@ -1039,11 +1045,13 @@ def action_print_survey(self, answer=None):
     def action_result_survey(self):
         """ Open the website page with the survey results view """
         self.ensure_one()
+        if not self.results_token:
+            self.results_token = self._get_random_token()
```

Comment:
I think there is a convention to keep names such as `get_default` so we know.
In this case, I think that this
```suggestion
            self.results_token = str(uuid.uuid4())  # see _get_default_random_token
```
 is better than
```py
    @api.model
    def _get_default_access_token(self):
        return self._get_random_token()

    @api.model
    def _get_random_token(self):
        return str(uuid.uuid4())
```